### DISCUSSION TITLE
Multiple Model Instance

### DISCUSSION THREAD

**Discussion by yuchenxie4645** [open-webui/open-webui#6620] [General]
I have a multiple user openwebui instance. I personally host a personally trained model in my webui along with 500 other models. I want to know how I can host multiple instances of one model. For example, I want it so that 3 people can use the llama 405b model at the same time, not having to wait for one generation to finish then continue to the second one.

**Comment by silentoplayz** [comment]
In the Ollama 0.2 update, concurrency management was significantly enhanced. To leverage these improvements, consider the following environment variables that can be set or adjusted when starting the Ollama server:

- **OLLAMA_NUM_PARALLEL**: This setting specifies the maximum number of parallel requests that a model can process at any given time. Set this environment variable before starting the server to optimize according to resource availability.

- **OLLAMA_MAX_LOADED_MODELS**: This allows you to control how many models can be loaded simultaneously in memory, which can help with resource management.

- **OLLAMA_MAX_QUEUE**: This option determines how many requests Ollama will queue when it’s busy, ensuring that incoming requests are handled efficiently without overwhelming the system.

By setting and or adjusting these particular environment variables, you can cater to your particular hardware configuration and expected load.

**Reply by yuchenxie4645** [reply]
Ok wait, how do I set these? 
![image](https://github.com/user-attachments/assets/5f38d596-04bb-4a30-8ef2-d5976e8b3bcc)
This is all I have.

**Reply by DesertCookie** [reply]
> Ok wait, how do I set these?

You have to set this on the Ollama server, not the Open WebUI. With Docker, you would set them as environment variables. In Windows you'd do the same via the env variables GUI.
